File: 0242-valid-anagram/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        Map<Character, Integer> counts = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            counts.put(c, counts.getOrDefault(c, 0) + 1);
        }

        for (int i = 0; i < t.length(); i++) {
            char c = t.charAt(i);
            if (!counts.containsKey(c)) {
                return false;
            }

            counts.put(c, counts.get(c) - 1);
            if (counts.get(c) == 0) {
                counts.remove(c);
            }
        }

        return counts.isEmpty();
    }
}

File: 0242-valid-anagram/solution.cpp
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }

        unordered_map<char, int> counts;
        for (char c : s) {
            counts[c]++;
        }

        for (char c : t) {
            if (!counts.count(c)) {
                return false;
            }

            counts[c]--;
            if (counts[c] == 0) {
                counts.erase(c);
            }
        }

        return counts.empty();
    }
};